teps/0056-pipelines-in-pipelines.md

```diff
@@ -0,0 +1,230 @@
+- Composing `Tasks` in `Tasks`, which is addressed in [TEP-0044 Composing Tasks with Tasks](https://github.com/tektoncd/community/pull/316)
+
+### Use Cases
+
+<!--
+Describe the concrete improvement specific groups of users will see if the
+Motivations in this doc result in a fix or feature.
+
+Consider both the user's role (are they a Task author? Catalog Task user?
+Cluster Admin? etc...) and experience (what workflows or actions are enhanced
+if this problem is solved?).
+-->
+
+#### Reusability and Composability
+
+As a `Pipeline` author, I need to define a set of `Tasks` as a complete unit of execution that I can share across `Pipelines`.
+
+For example, I can define and distribute `linting` set of `Tasks` and `testing` set of `Tasks`:
+```
+        linting                     
+           |                           
+           v
+        testing
+```
+
+where `linting` is made up of `lint` and `report-linter-output` tasks:
+``` 
+          lint                     
+           |                           
+           v
+   report-linter-output 
+```
+
+and `testing` is made up of `unit-tests`, `integration-tests` and `report-test-results` `Tasks`:
+```                           
+       unit-tests
+           |                           
+           v                           
+   integration-tests 
+           |
+           v 
+  report-test-results
+```
+
+#### Failure Strategies
+
+As a `Pipeline` author, I need to decouple failures in unrelated sets of `Tasks`.
+
+For example, I can design a `Pipeline` where `lint`might fail, but the `unit-tests`, `integration-tests` and `report-test-results` will continue executing so that I can get the test results without a rerun of the `Pipeline`:
+
+```
+          lint                     unit-tests
+           |                           |
+           v                           v
+   report-linter-output        integration-tests 
+                                       |
+                                       v 
+                              report-test-results
+```
+
+## Requirements
+
+<!--
+Describe constraints on the solution that must be met. Examples might include
+performance characteristics that must be met, specific edge cases that must
+be handled, or user scenarios that will be affected and must be accommodated.
+-->
+
+- Users should be able to define and distribute a set of `Tasks` as a complete unit of execution
+- Users should be able decouple failures in unrelated sets of `Tasks`
+- Users should be able to pass inputs (such as `Parameters`) from the main-`Pipeline` to the sub-`Pipeline`
+- Users should be able to access outputs (`Results`) from the sub-`Pipeline` in the main-`Pipeline`
+- Users should be able to access the status (`ConditionSucceeded`) of the sub-`Pipeline` in the main-`Pipeline`
+- Users should be able to propagate actions from the main-`Pipeline` to the sub-`Pipeline`, such as deletion and cancellation
```

Comment:
🎉  very exciting 🎉  I will add one more requirement of an universal access (doesnt have to be addressed in this TEP though) i.e. access the status of the main-pipeline in sub-pipelines 😱